autode/bracket/dhs.py

```diff
@@ -0,0 +1,728 @@
+            f" pair with {self._engrad_method}"
+        )
+
+        from autode.calculations import Calculation
+
+        engrad_calc = Calculation(
+            name=f"{img.name}_engrad",
+            molecule=img,
+            method=self._engrad_method,
+            keywords=self._engrad_method.keywords.grad,
+            n_cores=self._n_cores,
+        )
+        engrad_calc.run()
+        engrad_calc.clean_up(force=True, everything=True)
+
+        if img.energy is None:
+            raise CalculationException("Energy/gradient calculation failed")
+
+        # update coord
+        coord.e = img.energy.to("Ha")
+        coord.update_g_from_cart_g(img.gradient.to("Ha/ang"))
+
+
+class DHS(BaseBracketMethod):
+    """
+    Dewar-Healy-Stewart method for finding transition states,
+    from the reactant and product structures
+    """
+
+    def __init__(
+        self,
+        initial_species: "Species",
+        final_species: "Species",
+        step_size: Union[Distance, float] = Distance(0.1, "ang"),
+        **kwargs,
+    ):
+        """
+        Dewar-Healy-Stewart method to find transition states.
+
+        1) The step size is 0.1 which is quite conservative, so may want
+        to increase that if convergence is slow; 2) The distance tolerance
+        should not be lowered any more than 1.0 Angstrom as DHS is unstable
+        when the distance is low, and there is a tendency for one image to
+        jump over the barrier
+
+        Args:
+            initial_species: The "reactant" species
+
+            final_species: The "product" species
+
+            step_size: The size of the DHS step taken along
+                        the linear path between reactant and
+                        product
+
+        Keyword Args:
+
+            maxiter: Maximum number of en/grad evaluations
+
+            dist_tol: The distance tolerance at which DHS will
+                      stop, values less than 1.0 Angstrom are not
+                      recommended.
+
+            gtol: Gradient tolerance for the optimiser micro-iterations
+                  in DHS (Hartree/angstrom)
+
+            cineb_at_conv: Whether to run CI-NEB calculation from the end
+                           points after the DHS is converged
+        """
+        super().__init__(initial_species, final_species, **kwargs)
+
+        # imgpair is only used for storing the points here
+        self.imgpair = DHSImagePair(initial_species, final_species)
+
+        # DHS needs to keep an extra reference to the calculation method
+        self._method = None
+
+        self._step_size = Distance(abs(step_size), "ang")
+        assert self._step_size < self.imgpair.dist
+
+        # NOTE: In DHS the micro-iterations are done separately, in
+        # an optimiser, so to keep track of the actual number of
+        # en/grad calls, this local variable is used
+        self._current_microiters: int = 0
+
+    @property
+    def _method_name(self):
+        return "DHS"
```

Comment:
@t-young31  I made the _method_name a class attribute now. It's just used for printing the logging messages. I made it a variable so that I don't have to retype the logger messages for each method, and I can just use generic lines with the name of the method supplied in subclasses. For example in _log_convergence() method and in many other places with logging. Could you please take a look at the new code and let me know if it's fine?

Reply:
would it be okay to use the class name? (using the `__name__` attr)

Reply:
@t-young31 It could work, but I was using the _method_name mainly for printing the logger messages. And the class name cannot have hyphens so it would be difficult to write DHS-GS or i-EIP or something similar. For DHS and BITSS would be fine. What if I change the property to `_name` only?

Reply:
fair point. I'd vote a `_name` class attribute then – e.g.

```python
class DHS(BaseBracketMethod):
    """
    Dewar-Healy-Stewart method for finding transition states,
    from the reactant and product structures
    """

    _name = "DHS"

    # rest of the class...
```

Reply:
@t-young31 I think your original suggestion of using class name directly was better so I used type(self).__name__ (wrapped by a property).